### PR TITLE
DOP-2731-3: Send manifests to docs-search-indexes-test

### PR DESCRIPTION
**Pull request by mango-db**
[DOP-2731](https://jira.mongodb.org/browse/DOP-2731). Moves the manifest generation workflow into the manifestJobHandler. Builds to `docs-search-indexes-test`, with folder `prd` for  builds queued by `\deploy` command, and folder `preprd` for builds queued by `\test-deploy` commands.

### CODE REVIEW

File: src/job/manifestJobHandler.ts [hunk 1]
```diff
@@ -40,16 +43,47 @@ export class ManifestJobHandler extends JobHandler {
...
+    const b = this._config.get<string>('searchIndexBucket') ?? 'docs-search-indexes-test';
+    // /deploy -> send to /prd folder. /test-deploy -> send to /preprd folder
+    const env = this._config.get<string>('env');
+    // Note: mut-index automatically prepends 'search-indexes/' to the folder.
+    const f = this._config.get<string>('searchIndexFolder')[env] ?? 'fallback-folder';
+    this.logger.info(this.currJob._id, `Manifest attempt to upload to bucket: ${b}, folder: ${f}`);
+    // Due to the dual existence of prefixes, check for both for redundancy
+    const maP = this.currJob.manifestPrefix ?? this.currJob.payload.manifestPrefix;
+    const muP = this.currJob.mutPrefix ?? this.currJob.payload.mutPrefix;
+    const url = this.currJob.payload.url;
```
Comment (rkhullar-mongodb) [inline src/job/manifestJobHandler.ts:55]: nit: totally optional but i'd suggest more indicative variables names for `b` `f` `maP` and `muP`

**Review by rkhullar-mongodb — APPROVED** [review]
looks good to me!

File: src/job/manifestJobHandler.ts (hunk 1 above)
Comment (mango-db) [reply src/job/manifestJobHandler.ts:55]: Haha, yes, this probably could be better... I went this route to keep the actual `mut-index` command somewhat condensed. Going to leave it

File: src/job/manifestJobHandler.ts [hunk 2]
```diff
@@ -40,16 +43,47 @@ export class ManifestJobHandler extends JobHandler {
 
   // TODO: Make this a non-state-mutating function, e.g. return the deployCommands?
```
Comment (casthewiz) [inline src/job/manifestJobHandler.ts:44]:
```suggestion
  // TODO: Make this a non-state-mutating function, e.g. return the deployCommands?
  // TODO: Separate logic for composing mut-index string into separate helper function? 
```

**Review by casthewiz — APPROVED** [review]
I think this LGTM, although I lack full context around the implications of switching from `python3 test-mut-script.py` to `mut-index`

**Comment by mango-db** [comment]
> I think this LGTM, although I lack full context around the implications of switching from `python3 test-mut-script.py` to `mut-index`

Basically, I realized that `mut-index` was already being loaded and decided to use it directly

**Pull request merged**